Explorer/Assets/DCL/UI/ColorPicker/ColorPickerController.cs

```diff
@@ -1,7 +1,172 @@
+using DCL.AvatarRendering.Wearables.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Pool;
+using Object = UnityEngine.Object;
+
 namespace DCL.UI
 {
     public class ColorPickerController
     {
-        
+        private const float INCREMENT_AMOUNT = 0.1f;
+        public event Action<Color, string> OnColorChanged;
+
+        private readonly ColorPickerView view;
+        private readonly ColorPresetsSO hairColors;
+        private readonly ColorPresetsSO eyesColors;
+        private readonly ColorPresetsSO bodyshapeColors;
+        private readonly IObjectPool<ColorToggleView> colorTogglesPool;
+        private readonly List<ColorToggleView> usedColorToggles = new ();
+        private string currentCategory = "";
+
+        private Color hairsColor;
+        private Color eyesColor;
+        private Color bodyshapeColor;
+
+        public ColorPickerController(ColorPickerView view, ColorToggleView colorToggle, ColorPresetsSO hairColors, ColorPresetsSO eyesColors, ColorPresetsSO bodyshapeColors)
+        {
+            this.view = view;
+            this.hairColors = hairColors;
+            this.eyesColors = eyesColors;
+            this.bodyshapeColors = bodyshapeColors;
+
+            colorTogglesPool = new ObjectPool<ColorToggleView>(
+                () => Object.Instantiate(colorToggle, view.ColorPresetsParent),
+                actionOnGet:(toggle) => toggle.gameObject.SetActive(true),
+                actionOnRelease:(toggle) => toggle.gameObject.SetActive(false));
+
+            view.SliderHue.Slider.onValueChanged.AddListener(_ => SetColor());
+            view.SliderHue.IncreaseButton.onClick.AddListener(() => ChangeProperty("hue", INCREMENT_AMOUNT));
+            view.SliderHue.DecreaseButton.onClick.AddListener(() => ChangeProperty("hue", -INCREMENT_AMOUNT));
+
+            view.SliderSaturation.Slider.onValueChanged.AddListener(_ => SetColor());
+            view.SliderSaturation.IncreaseButton.onClick.AddListener(() => ChangeProperty("sat", INCREMENT_AMOUNT));
+            view.SliderSaturation.DecreaseButton.onClick.AddListener(() => ChangeProperty("sat", -INCREMENT_AMOUNT));
+
+            view.SliderValue.Slider.onValueChanged.AddListener(_ => SetColor());
+            view.SliderValue.IncreaseButton.onClick.AddListener(() => ChangeProperty("val", INCREMENT_AMOUNT));
+            view.SliderValue.DecreaseButton.onClick.AddListener(() => ChangeProperty("val", -INCREMENT_AMOUNT));
+
+            view.ToggleButton.onClick.AddListener(() => TogglePanel());
+        }
+
+        public void SetCurrentColor(Color newColor, string category)
```

Comment:
It's not a problem because this sets the colors locally but then the update of the sliders is done when picking a category (because the panel is reused per category)